tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,89 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 *
+	 * @dataProvider data_get_block_theme_folders
+	 *
+	 * @param string   $theme    The theme's stylesheet.
+	 * @param string[] $expected The expected associative array of block theme folders.
+	 * @param string   $message The message for when an assertion fails.
+	 */
+	public function test_get_block_theme_folders( $theme, $expected, $message ) {
+		$wp_theme   = wp_get_theme( $theme );
+		$reflection = new ReflectionMethod( $wp_theme, 'cache_get' );
+		$reflection->setAccessible( true );
+		$wp_theme->cache_delete(); // Clear cache.
+
+		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
+		$theme_cache  = $reflection->invoke( $wp_theme, 'theme' );
+		$cached_value = $theme_cache['block_template_folders'];
+		$reflection->setAccessible( false );
+
+		$this->assertSame( $expected, $cached_value, 'Cache - ' . $message );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_get_block_theme_folders() {
+		return array(
+			'block-theme'                       => array(
+				'block-theme',
+				array(
+					'wp_template'      => 'templates',
+					'wp_template_part' => 'parts',
+				),
+				'For block-theme, expected `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
+			),
+			'block-theme-deprecated-path'       => array(
+				'block-theme-deprecated-path',
+				array(
+					'wp_template'      => 'block-templates',
+					'wp_template_part' => 'block-template-parts',
+				),
+				'For block-theme-deprecated-path, expected `wp_template` to be `block-templates` and `block-template-parts` to be `parts`.',
+			),
+			'block-theme-child'                 => array(
+				'block-theme-child',
+				array(
+					'wp_template'      => 'templates',
+					'wp_template_part' => 'parts',
+				),
+				'For block-theme-child, expected `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
+			),
+			'block-theme-child-deprecated-path' => array(
+				'block-theme-child-deprecated-path',
+				array(
+					'wp_template'      => 'block-templates',
+					'wp_template_part' => 'block-template-parts',
+				),
+				'For block-theme-child-deprecated-path, expected `wp_template` to be `block-templates` and `block-template-parts` to be `parts`.',
+			),
+			'this-is-an-invalid-theme'          => array(
+				'this-is-an-invalid-theme',
+				array(
+					'wp_template'      => 'templates',
+					'wp_template_part' => 'parts',
+				),
+				'For invalid theme, expected default values `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
+			),
+			'null'                              => array(
+				null,
+				array(
+					'wp_template'      => 'templates',
+					'wp_template_part' => 'parts',
+				),
+				'For no theme, expected default values `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
+			),
+		);
+	}
```

Comment:
```suggestion
		$this->assertSame( $expected, $cached_value, $message );
	}

	/**
	 * Data provider.
	 *
	 * @return array[]
	 */
	public function data_get_block_theme_folders() {
		return array(
			'block-theme'                       => array(
				'block-theme',
				array(
					'wp_template'      => 'templates',
					'wp_template_part' => 'parts',
				),
				'Cache - For block-theme, expected `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
			),
			'block-theme-deprecated-path'       => array(
				'block-theme-deprecated-path',
				array(
					'wp_template'      => 'block-templates',
					'wp_template_part' => 'block-template-parts',
				),
				'Cache - For block-theme-deprecated-path, expected `wp_template` to be `block-templates` and `block-template-parts` to be `parts`.',
			),
			'block-theme-child'                 => array(
				'block-theme-child',
				array(
					'wp_template'      => 'templates',
					'wp_template_part' => 'parts',
				),
				'Cache - For block-theme-child, expected `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
			),
			'block-theme-child-deprecated-path' => array(
				'block-theme-child-deprecated-path',
				array(
					'wp_template'      => 'block-templates',
					'wp_template_part' => 'block-template-parts',
				),
				'Cache - For block-theme-child-deprecated-path, expected `wp_template` to be `block-templates` and `block-template-parts` to be `parts`.',
			),
			'this-is-an-invalid-theme'          => array(
				'this-is-an-invalid-theme',
				array(
					'wp_template'      => 'templates',
					'wp_template_part' => 'parts',
				),
				'Cache - For invalid theme, expected default values `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
			),
			'null'                              => array(
				null,
				array(
					'wp_template'      => 'templates',
					'wp_template_part' => 'parts',
				),
				'Cache - For no theme, expected default values `wp_template` to be `templates` and `wp_template_part` to be `parts`.',
			),
		);
	}
```

Reply:
@spacedmonkey  `test_get_block_theme_folders` has 2 assertion using same message. So I was pre-pending "Cached - " in the assertion for cache and using the message directly for the first assertion.

Reply:
Removed "Cache - " instead. 
We kind of get the line number so shouldn't be an issue.